[Docs] Add guide for debugging PHP in WebAssembly with Xdebug (#2949)
WordPress/wordpress-playground: packages/docs/site/docs/developers/07-xdebug/03-xdebug-on-php-wasm.md

---
title: Debugging with Xdebug in PHP WASM CLI
slug: /developers/xdebug/xdebug-on-php-wasm
description: Debug PHP in WebAssembly with Xdebug using VS Code, Cursor, or PhpStorm. Set breakpoints and inspect variables.
---

# Debugging with Xdebug in PHP WASM CLI

The **php-wasm CLI** supports Xdebug, allowing developers to debug PHP code running within the WebAssembly environment. Set breakpoints, step through code execution, inspect variables, and modify them at runtime.

PHP WASM CLI debugs any PHP code—you don't need WordPress. This works for all PHP developers.

## Prerequisites

You need an IDE that supports Xdebug and its configuration file.

> -   **A configuration file:** For example, `.vscode/launch.json` for VS Code-based IDEs, or a Run/Debug Configuration created via **Run ▸ Edit Configurations…** in PhpStorm

Reviewer: ```suggestion
-   **A configuration file:** For example, `.vscode/launch.json` for VS Code-based IDEs and `.idea/workspace.xml` for PhpStorm
```

Reviewer: Copilot changed this while fixing the image url.

Reviewer: I didn't need a `.vscode/launch.json` file. The _Listen for Xdebug_ button was available in the Debug panel without explicitly configuring a debugger.

-   **VS Code / Cursor:** We recommend the official [PHP Debug extension](https://marketplace.visualstudio.com/items?itemName=xdebug.php-debug).
-   **PhpStorm:** supports Xdebug natively.

## Usage Guide

### 1. Create or select your PHP file

This example demonstrates Xdebug debugging:

```php
<?php

$test = 42; // Set a breakpoint on this line

echo "Output!\n";

function test( $output ) {
    echo "Hello Xdebug World!\n" . $output . "\n";
}

test($test);
```

### 2. Configure your IDE

Before running the script, prepare your IDE to listen for the Xdebug connection.

#### **VS Code / Cursor Instructions**

1.  Ensure you have installed the **PHP Debug** extension.
2.  Open the **Run and Debug** panel on the left sidebar.

> 3.  Select **"PHP.wasm CLI - Listen for Xdebug"** from the configuration dropdown.

Reviewer: There is no _PHP.wasm CLI_ in the debug sidebar, just _Listen for Xdebug_. 

```suggestion
3.  Click **"Listen for Xdebug"** from the configuration dropdown.
```
